CHANGELOG.md

```diff
@@ -1,8 +1,26 @@
-# Release Notes for Kubernetes 1.2.0
+<!--THIS TOC NEEDS TO BE GENERATED EVERY TIME THIS FILE IS UPDATED -->
```

Comment:
Oh, I just wrote this quickly to generate it.  The "GENERATED_TOC" comment in the link you sent seems to indicate something does this automagically somehow/somewhere.  I'll take a look.
```
#!/bin/bash
#
# Markdown TOC
#
# Throws out toplevel (#) heading assuming it's the title
firstheading="##"
if [[ "$1" == --include-title ]]; then
  firstheading="#"
  shift
fi
file=$1

echo "Table of Contents"
echo "================="
while read level heading; do
  indent="$(echo $level |sed -e "s,^$firstheading,,g" -e 's,#,  ,g')"
  # fix anchor
  anchor=${heading,,}
  anchor=${anchor// /-}
  anchor=${anchor//[\?\*\,\/()]/}
  echo "$indent* [$heading](#$anchor)"
done < <(egrep "^$firstheading" $file)
```